pkg/contract-manager/reservation_controller.go

```diff
@@ -47,13 +56,318 @@ type ReservationReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.2/pkg/reconcile
 func (r *ReservationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	klog.Info("Reconciling Reservation")
+	log := ctrl.LoggerFrom(ctx, "discovery", req.NamespacedName)
+	ctx = ctrl.LoggerInto(ctx, log)
 
-	// TODO(user): your logic here
+	// var contract *reservationv1alpha1.Contract
+
+	var transaction *Transaction
+	var reservation reservationv1alpha1.Reservation
+	if err := r.Get(ctx, req.NamespacedName, &reservation); client.IgnoreNotFound(err) != nil {
+		klog.Errorf("Error when getting Reservation %s before reconcile: %s", req.NamespacedName, err)
+		return ctrl.Result{}, err
+	} else if err != nil {
+		klog.Infof("Reservation %s not found, probably deleted", req.NamespacedName)
+		return ctrl.Result{}, nil
+	}
+
+	if reservation.Status.Phase.Phase != nodecorev1alpha1.PhaseSolved &&
+		reservation.Status.Phase.Phase != nodecorev1alpha1.PhaseTimeout &&
+		reservation.Status.Phase.Phase != nodecorev1alpha1.PhaseFailed &&
+		reservation.Status.Phase.Phase != nodecorev1alpha1.PhaseRunning &&
+		reservation.Status.Phase.Phase != nodecorev1alpha1.PhaseIdle {
+
+		klog.Infof("Reservation %s started", reservation.Name)
+		reservation.Status.Phase.StartTime = time.Now().Format(time.RFC3339)
```

Comment:
do you remember my suggestion about a function `getTimeNow`?

Reply:
Fixed, function moved to `utils/common`